File: src/main/java/com/webapp/dto/MenuEntryDto.java
package com.webapp.dto;

import com.webapp.model.MenuEntry;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import java.io.Serializable;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@ToString
public class MenuEntryDto implements Serializable {

    private String id;
    private String parentId;
    private String url;
    private String name;
    private List<String> roles;

    public MenuEntryDto(MenuEntry menuEntry) {
        this.id = menuEntry.getId();
        this.parentId = menuEntry.getParentId();
        this.url = menuEntry.getUrl();
        this.name = menuEntry.getName();
        this.roles = menuEntry.getRoles();
    }

    public MenuEntryDto(String url, String name, List<String> roles) {
        this.url = url;
        this.name = name;
        this.roles = roles;
    }
}

File: addressbook-common/src/main/kotlin/com/addressbook/dto/MenuEntryDto.kt
package com.addressbook.dto

import com.addressbook.model.MenuEntry
import java.io.Serializable

class MenuEntryDto(menuEntry: MenuEntry? = null) : Serializable {

    var id: String? = null
    var parentId: String? = null
    var url: String? = null
    var name: String? = null
    var roles: List<String>? = null

    init {
        this.id = menuEntry?.id
        this.parentId = menuEntry?.parentId
        this.url = menuEntry?.url
        this.name = menuEntry?.name
        this.roles = menuEntry?.roles
    }

    constructor(url: String, name: String, roles: List<String>) : this() {
        this.url = url
        this.name = name
        this.roles = roles
    }
}
